spec/rubygems_ranges_spec.rb

```diff
@@ -213,54 +220,48 @@
       ['<= 2.0.0', '<=2.0.0'],
       ['<=  2.0.0', '<=2.0.0'],
       ['<    2.0.0', '<2.0.0'],
-      ['<	2.0.0', '<2.0.0'],
-      ['>=0.1.97', '>=0.1.97'],
+      ["<\t2.0.0", '<2.0.0'],
       ['>=0.1.97', '>=0.1.97'],
       ['0.1.20 || 1.2.4', '0.1.20||1.2.4'],
       ['>=0.2.3 || <0.0.1', '>=0.2.3||<0.0.1'],
-      ['>=0.2.3 || <0.0.1', '>=0.2.3||<0.0.1'],
-      ['>=0.2.3 || <0.0.1', '>=0.2.3||<0.0.1'],
-      ['||', '||'],
-      ['2.x.x', '>=2.0.0 <3.0.0'],
-      ['1.2.x', '>=1.2.0 <1.3.0'],
-      ['1.2.x || 2.x', '>=1.2.0 <1.3.0||>=2.0.0 <3.0.0'],
-      ['1.2.x || 2.x', '>=1.2.0 <1.3.0||>=2.0.0 <3.0.0'],
+      ['||', '*'],
+      ['2.x.x', '>=2.0.0 <3.0.0-0'],
+      ['1.2.x', '>=1.2.0 <1.3.0-0'],
+      ['1.2.x || 2.x', '>=1.2.0 <1.3.0-0||>=2.0.0 <3.0.0-0'],
       ['x', '*'],
-      ['2.*.*', '>=2.0.0 <3.0.0'],
-      ['1.2.*', '>=1.2.0 <1.3.0'],
-      ['1.2.* || 2.*', '>=1.2.0 <1.3.0||>=2.0.0 <3.0.0'],
-      ['*', '*'],
-      ['2', '>=2.0.0 <3.0.0'],
-      ['2.3', '>=2.3.0 <2.4.0'],
+      ['2.*.*', '>=2.0.0 <3.0.0-0'],
+      ['1.2.*', '>=1.2.0 <1.3.0-0'],
+      ['1.2.* || 2.*', '>=1.2.0 <1.3.0-0||>=2.0.0 <3.0.0-0'],
+      ['2', '>=2.0.0 <3.0.0-0'],
+      ['2.3', '>=2.3.0 <2.4.0-0'],
       ['~2.4', '>=2.4.0 <3.0.0'],
-      ['~>3.2.1', '>=3.2.1 <3.3.0'],
-      ['~1', '>=1.0.0 <2.0.0'],
-      ['~>1', '>=1.0.0 <2.0.0'],
-      ['~> 1', '>=1.0.0 <2.0.0'],
+      ['~>3.2.1', '>=3.2.1 <3.3.0-0'],
+      ['~1', '>=1.0.0 <2.0.0-0'],
```

Comment:
It looks like some of the stuff above was duplicated, but some test cases here are changed. I'd feel more comfortable if the cases were in addition to the existing ones ones (obviously removing dupes is fine).

Reply:
@jsonperl As I understand, the existing ones were created in a similar way: copied from node-semver. I assume copy-paste-replace is the only way to keep them in sync with node-semver 🙄